### ISSUE TITLE
proto3 hasMETHOD always generated 

### ISSUE THREAD

**Issue by kutoman** [google/protobuf.dart#767]
I'm not sure whether this is an issue or intended. In the generated dart code the `has..` method is always generated for a corresponding `message` field regardless of being labeled as `optional` or not. Does it make sense to generate those methods for none-optional fields?

**Comment by osa1** [comment]
Yes. We have a TODO about this in the code: https://github.com/google/protobuf.dart/blob/ba29983968de5b54196c6c2a5cff1afbdda24ae2/protoc_plugin/lib/src/protobuf_field.dart#L129-L140 this was added in 13198471df7f0bbe60b264f061cfe011cf29e89b which implemented proto3 `optional` syntax.

**Comment by kutoman** [comment]
oh I see. I didn't check the generator's code
